internal/pki/pki_test.go

```diff
@@ -27,6 +27,38 @@ import (
 	"gotest.tools/v3/assert"
 )
 
+func TestCertificateEqual(t *testing.T) {
+	zero := Certificate{}
+	assert.DeepEqual(t, zero, zero)
+
+	root := NewRootCertificateAuthority()
+	assert.NilError(t, root.Generate())
+	assert.DeepEqual(t, root.Certificate, root.Certificate)
+
+	assert.Assert(t, !root.Certificate.Equal(zero))
+	assert.Assert(t, !zero.Equal(*root.Certificate))
+
+	other := NewRootCertificateAuthority()
+	assert.NilError(t, other.Generate())
+	assert.Assert(t, !root.Certificate.Equal(*other.Certificate))
+}
+
+func TestPrivateKeyEqual(t *testing.T) {
+	zero := PrivateKey{}
+	assert.DeepEqual(t, zero, zero)
+
+	root := NewRootCertificateAuthority()
+	assert.NilError(t, root.Generate())
+	assert.DeepEqual(t, root.PrivateKey, root.PrivateKey)
+
+	assert.Assert(t, !root.PrivateKey.Equal(zero))
+	assert.Assert(t, !zero.Equal(*root.PrivateKey))
+
+	other := NewRootCertificateAuthority()
+	assert.NilError(t, other.Generate())
+	assert.Assert(t, !root.PrivateKey.Equal(*other.PrivateKey))
```

Comment:
Could we add test cases where the Certificates / PrivateKeys were equal?

Reply:
That's L32, L36, L48, and L52. I'll rearrange to show that the `Equal` method works with and without `DeepEqual`.

Without the `Equal` method, `assert.DeepEqual` tries to call [`ecdsa.PrivateKey.Equal`](https://pkg.go.dev/crypto/ecdsa#PrivateKey.Equal) which panics when the receiver is nil (our zero value).

On other PrivateKey values, without the `Equal` method, `assert.DeepEqual` fails saying "cannot handle unexported field at {*pki.PrivateKey}.marshalECPrivateKey".

Reply:
Ah, ok, I think this comment helps explain that, so maybe this comment is all you need?

Reply:
Amended in af07b40d0.